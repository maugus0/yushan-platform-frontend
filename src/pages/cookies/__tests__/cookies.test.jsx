// src/pages/cookies/cookies.test.jsx
import { render, screen } from '@testing-library/react';
import CookiePolicy from '../cookies';

describe('CookiePolicy page', () => {
  test('renders main header and updated date', () => {
    const { container } = render(<CookiePolicy />);
    // main heading - select the actual H1 to avoid matching other occurrences of the phrase
    const h1 = container.querySelector('h1');
    expect(h1).toBeTruthy();
    expect(h1).toHaveTextContent(/Cookie Policy/i);
    // updated date text - scope to header element to avoid ambiguous matches
    const updated = container.querySelector('.cookies-updated');
    expect(updated).toBeTruthy();
    expect(updated).toHaveTextContent(/Last updated:/i);
    // container sanity
    expect(container.querySelector('.cookies-page')).toBeTruthy();
  });

  test('renders section headings and usage list items', () => {
    render(<CookiePolicy />);
    expect(screen.getByText(/What Are Cookies\?/i)).toBeInTheDocument();
    expect(screen.getByText(/How We Use Cookies/i)).toBeInTheDocument();
    expect(screen.getByText(/Types of Cookies We Use/i)).toBeInTheDocument();
    expect(screen.getByText(/Managing Your Cookie Preferences/i)).toBeInTheDocument();
    // check some list items
    expect(screen.getByText(/Keeping you logged in to your account/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Remembering your reading preferences and settings/i)
    ).toBeInTheDocument();
    expect(screen.getByText(/Google Analytics:/i)).toBeInTheDocument();
  });

  test('renders table with expected columns and rows', () => {
    const { container } = render(<CookiePolicy />);
    // AntD Table renders a semantic table
    const table = container.querySelector('table');
    expect(table).toBeTruthy();

    // Check column headers exist (use text)
    const thead = table.querySelector('thead');
    expect(thead).toBeTruthy();
    const headerText = thead.textContent;
    expect(headerText).toMatch(/Cookie Type/i);
    expect(headerText).toMatch(/Purpose/i);
    expect(headerText).toMatch(/Duration/i);
    expect(headerText).toMatch(/Examples/i);

    // Data rows are in tbody
    const tbodyRows = container.querySelectorAll('tbody tr');
    // Expect number of data rows equals 4 (based on component data)
    expect(tbodyRows.length).toBe(4);

    // Validate first row cell contents
    const firstRowCells = tbodyRows[0].querySelectorAll('td');
    // first column: cookie name
    expect(firstRowCells[0].textContent).toContain('Essential Cookies');
    // second column: purpose snippet
    expect(firstRowCells[1].textContent).toMatch(/Authentication, security/i);
    // third column: duration
    expect(firstRowCells[2].textContent).toMatch(/30 days|Session/i);
    // fourth column: examples
    expect(firstRowCells[3].textContent).toMatch(/auth_token|session_id/i);
  });

  test('renders contact information and additional paragraphs', () => {
    render(<CookiePolicy />);
    // Contact email and address present
    expect(screen.getByText(/privacy@yushan.com/i)).toBeInTheDocument();
    expect(screen.getByText(/Yushan Interactive Pte. Ltd., Singapore/i)).toBeInTheDocument();
    // Some paragraph contents
    expect(screen.getByText(/Cookies are small text files/i)).toBeInTheDocument();
    expect(screen.getByText(/We may update this Cookie Policy/i)).toBeInTheDocument();
  });

  test('table elements are queryable via role and snapshot', () => {
    const { container } = render(<CookiePolicy />);
    // role-based table query
    const roleTable = screen.getByRole('table');
    expect(roleTable).toBe(container.querySelector('table'));

    // snapshot to cover rendering markup
    expect(container).toMatchSnapshot();
  });
});

// Mock AntD responsive utilities to avoid runtime errors in JSDOM
jest.mock('antd/lib/grid/hooks/useBreakpoint', () => () => ({
  xs: true,
  sm: true,
  md: true,
  lg: true,
  xl: true,
  xxl: true,
}));
jest.mock('antd/lib/_util/responsiveObserver', () => ({
  __esModule: true,
  default: () => ({
    subscribe: (listener) => {
      try {
        listener({ xs: true, sm: true, md: true, lg: true, xl: true, xxl: true });
      } catch (e) {
        void e;
      }
      return () => {};
    },
    unsubscribe: () => {},
    dispatch: () => {},
  }),
}));
