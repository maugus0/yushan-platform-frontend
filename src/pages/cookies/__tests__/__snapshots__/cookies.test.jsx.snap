// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiePolicy page table elements are queryable via role and snapshot 1`] = `
<div>
  <div
    class="cookies-page"
  >
    <div
      class="cookies-container"
    >
      <div
        class="ant-row ant-row-center css-dev-only-do-not-override-1odpy5d"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-sm-22 ant-col-md-20 ant-col-lg-16 ant-col-xl-14 css-dev-only-do-not-override-1odpy5d"
        >
          <div
            class="ant-card ant-card-bordered cookies-card css-dev-only-do-not-override-1odpy5d"
          >
            <div
              class="ant-card-body"
            >
              <div
                class="cookies-header"
              >
                <h1
                  class="ant-typography cookies-title css-dev-only-do-not-override-1odpy5d"
                >
                  Cookie Policy
                </h1>
                <span
                  class="ant-typography cookies-updated css-dev-only-do-not-override-1odpy5d"
                >
                  Last updated: October 12, 2025
                </span>
              </div>
              <div
                class="ant-divider css-dev-only-do-not-override-1odpy5d ant-divider-horizontal"
                role="separator"
              />
              <div
                class="cookies-content"
              >
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  What Are Cookies?
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Cookies are small text files that are stored on your device when you visit our website. They help us provide you with a better experience by remembering your preferences and understanding how you use our platform.
                </div>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  How We Use Cookies
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Yushan uses cookies for various purposes, including:
                </div>
                <ul
                  class="cookies-list"
                >
                  <li>
                    Keeping you logged in to your account
                  </li>
                  <li>
                    Remembering your reading preferences and settings
                  </li>
                  <li>
                    Analyzing site traffic and user behavior
                  </li>
                  <li>
                    Personalizing content and recommendations
                  </li>
                  <li>
                    Improving our services and user experience
                  </li>
                </ul>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Types of Cookies We Use
                </h2>
                <div
                  class="cookies-table-wrapper"
                >
                  <div
                    class="ant-table-wrapper css-dev-only-do-not-override-1odpy5d"
                  >
                    <div
                      class="ant-spin-nested-loading css-dev-only-do-not-override-1odpy5d"
                    >
                      <div
                        class="ant-spin-container"
                      >
                        <div
                          class="ant-table ant-table-middle ant-table-bordered css-dev-only-do-not-override-1odpy5d"
                        >
                          <div
                            class="ant-table-container"
                          >
                            <div
                              class="ant-table-content"
                            >
                              <table
                                style="table-layout: auto;"
                              >
                                <colgroup>
                                  <col
                                    style="width: 20%;"
                                  />
                                  <col
                                    style="width: 35%;"
                                  />
                                  <col
                                    style="width: 20%;"
                                  />
                                  <col
                                    style="width: 25%;"
                                  />
                                </colgroup>
                                <thead
                                  class="ant-table-thead"
                                >
                                  <tr>
                                    <th
                                      class="ant-table-cell"
                                      scope="col"
                                    >
                                      Cookie Type
                                    </th>
                                    <th
                                      class="ant-table-cell"
                                      scope="col"
                                    >
                                      Purpose
                                    </th>
                                    <th
                                      class="ant-table-cell"
                                      scope="col"
                                    >
                                      Duration
                                    </th>
                                    <th
                                      class="ant-table-cell"
                                      scope="col"
                                    >
                                      Examples
                                    </th>
                                  </tr>
                                </thead>
                                <tbody
                                  class="ant-table-tbody"
                                >
                                  <tr
                                    class="ant-table-row ant-table-row-level-0"
                                    data-row-key="1"
                                  >
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Essential Cookies
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Authentication, security, and basic site functionality
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Session/30 days
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      auth_token, session_id
                                    </td>
                                  </tr>
                                  <tr
                                    class="ant-table-row ant-table-row-level-0"
                                    data-row-key="2"
                                  >
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Analytics Cookies
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Understanding how users interact with our platform
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      2 years
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Google Analytics cookies
                                    </td>
                                  </tr>
                                  <tr
                                    class="ant-table-row ant-table-row-level-0"
                                    data-row-key="3"
                                  >
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Preference Cookies
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Remembering user settings and preferences
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      1 year
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      theme_preference, language_setting
                                    </td>
                                  </tr>
                                  <tr
                                    class="ant-table-row ant-table-row-level-0"
                                    data-row-key="4"
                                  >
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Marketing Cookies
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      Delivering relevant advertisements and tracking campaigns
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      90 days
                                    </td>
                                    <td
                                      class="ant-table-cell"
                                    >
                                      ad_tracking, campaign_source
                                    </td>
                                  </tr>
                                </tbody>
                              </table>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Managing Your Cookie Preferences
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  You can control and manage cookies in several ways:
                </div>
                <ul
                  class="cookies-list"
                >
                  <li>
                    <span
                      class="ant-typography css-dev-only-do-not-override-1odpy5d"
                    >
                      <strong>
                        Browser Settings:
                      </strong>
                    </span>
                     Most browsers allow you to view, manage, and delete cookies through their settings.
                  </li>
                  <li>
                    <span
                      class="ant-typography css-dev-only-do-not-override-1odpy5d"
                    >
                      <strong>
                        Opt-out Tools:
                      </strong>
                    </span>
                     You can opt out of analytics cookies through Google Analytics' opt-out browser add-on.
                  </li>
                  <li>
                    <span
                      class="ant-typography css-dev-only-do-not-override-1odpy5d"
                    >
                      <strong>
                        Do Not Track:
                      </strong>
                    </span>
                     We respect Do Not Track signals where technically feasible.
                  </li>
                </ul>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Essential Cookies
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Some cookies are essential for our website to function properly. These cannot be disabled without affecting the basic functionality of our platform, such as user authentication and security features.
                </div>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Third-Party Cookies
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  We may use third-party services that set cookies on your device. These include:
                </div>
                <ul
                  class="cookies-list"
                >
                  <li>
                    <span
                      class="ant-typography css-dev-only-do-not-override-1odpy5d"
                    >
                      <strong>
                        Google Analytics:
                      </strong>
                    </span>
                     For website analytics and performance monitoring
                  </li>
                  <li>
                    <span
                      class="ant-typography css-dev-only-do-not-override-1odpy5d"
                    >
                      <strong>
                        Social Media:
                      </strong>
                    </span>
                     For social sharing functionality
                  </li>
                  <li>
                    <span
                      class="ant-typography css-dev-only-do-not-override-1odpy5d"
                    >
                      <strong>
                        Content Delivery Networks:
                      </strong>
                    </span>
                     For faster content delivery
                  </li>
                </ul>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Updates to This Policy
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  We may update this Cookie Policy from time to time to reflect changes in our practices or for other operational, legal, or regulatory reasons. We will notify you of any material changes by posting the updated policy on our website.
                </div>
                <h2
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  Contact Us
                </h2>
                <div
                  class="ant-typography css-dev-only-do-not-override-1odpy5d"
                >
                  If you have any questions about our use of cookies, please contact us at:
                  <br />
                  <span
                    class="ant-typography css-dev-only-do-not-override-1odpy5d"
                  >
                    <strong>
                      Email:
                    </strong>
                  </span>
                   privacy@yushan.com
                  <br />
                  <span
                    class="ant-typography css-dev-only-do-not-override-1odpy5d"
                  >
                    <strong>
                      Address:
                    </strong>
                  </span>
                   Yushan Interactive Pte. Ltd., Singapore
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
